src/codeedges.jl

```diff
@@ -299,23 +308,35 @@ function add_links!(target::Pair{Union{SSAValue,SlotNumber,GlobalRef},Links}, @n
             stmt = GlobalRef(cl.thismod, stmt)
         end
         push!(targetstore, stmt)
-        namestore = get(cl.namesuccs, stmt, nothing)
-        if namestore === nothing
-            cl.namesuccs[stmt] = namestore = Links()
-        end
+        namestore = get!(Links, cl.namesuccs, stmt)
         push!(namestore, targetid)
-    elseif isa(stmt, Expr) && stmt.head !== :copyast
-        stmt = stmt::Expr
-        arng = 1:length(stmt.args)
-        if stmt.head === :call
-            f = stmt.args[1]
-            if !@isssa(f) && !@issslotnum(f)
-                # Avoid putting named callees on the namestore
-                arng = 2:length(stmt.args)
+    elseif isa(stmt, Expr)
+        if stmt.head === :globaldecl
+            for i = 1:length(stmt.args)
+                a = stmt.args[i]
+                if a isa GlobalRef
+                    namestore = get!(Links, cl.namepreds, a) # TODO should this information be tracked in the separate `cl.namedecls` store?
+                    push!(namestore, targetid)
+                    if targetid isa SSAValue
+                        push!(namestore, SSAValue(targetid.id+1)) # +1 for :latestworld
+                    end
+                else
+                    add_links!(target, a, cl)
+                end
+            end
+        elseif stmt.head !== :copyast
+            stmt = stmt::Expr
```

Comment:
This line seems like it must date to an era when inference was not as capable, though obviously it does no harm.